src/lcmap/aardvark/landsat.clj

```diff
@@ -78,10 +78,12 @@
 (defn get-tile-spec
   "Search for a source and produce a response map."
   [ubid {params :params :as req}]
-  (log/debug "get tile-spec for '%s' with %s" ubid params)
-  (if-let [results (first (tile-spec/query (merge {:ubid ubid} params)))]
-    {:status 200 :body results}
-    {:status 404 :body nil}))
+  (log/debugf "get tile-spec for '%s' with %s" ubid params)
+  (let [query  (str "ubid:" (str/replace ubid #"/" " AND "))
+        result (first (index/result (index/search query)))]
+    (if (= (:ubid result) ubid)
```

Comment:
this is now necessary because if a user supplies LANDSAT_5/TM as a result instead of LANDSAT_5/TM/sr_band1 they will get a search result.